packages/store/types/tests/store.lint.ts

```diff
@@ -137,4 +146,120 @@ describe('[TEST]: Action Types', () => {
     assertType(() => store.selectSnapshot({ foo: 'bar' })); // $ExpectError
     assertType(() => store.selectSnapshot()); // $ExpectError
   });
+
+  it('should be correct detect type with store methods in component', () => {
+    @State<string[]>({
+      name: 'todo',
+      defaults: []
+    })
+    class TodoState {
+      @Selector() // $ExpectType (state: string[]) => string[]
+      public static todo(state: string[]): string[] {
+        return state;
+      }
+    }
+
+    class TestComponent {
+      public A1$: Observable<string[]> = this.select(TodoState.todo);
+      public B1$: Observable<number[]> = this.select(TodoState.todo); // $ExpectError
+      public C1$: Observable<number[]> = this.select<number[]>(TodoState.todo); // $ExpectError
+      public D1$: Observable<number[]> = this.select<any>(TodoState.todo);
+      public F1$: Observable<string[]> = this.select<string[]>(TodoState.todo);
+      public A2$: Observable<string[]> = this.selectOnce(TodoState.todo);
+      public B2$: Observable<number[]> = this.selectOnce(TodoState.todo); // $ExpectError
+      public C2$: Observable<number[]> = this.selectOnce<number[]>(TodoState.todo); // $ExpectError
+      public D2$: Observable<number[]> = this._store.selectOnce<any>(TodoState.todo);
+      public F2$: Observable<string[]> = this._store.selectOnce<string[]>(TodoState.todo);
+      public A3$: string[] = this.selectSnapshot(TodoState.todo);
+      public B3$: number[] = this.selectSnapshot(TodoState.todo); // $ExpectError
+      public C3$: number[] = this.selectSnapshot<number[]>(TodoState.todo); // $ExpectError
+      public D3$: number[] = this.selectSnapshot<any>(TodoState.todo);
+      public F3$: string[] = this.selectSnapshot<string[]>(TodoState.todo);
+
+      constructor(private readonly _store: Store) {}
+
+      private get selectSnapshot() {
+        return this._store.selectSnapshot;
+      }
+
+      private get selectOnce() {
+        return this._store.selectOnce;
+      }
+
+      private get select() {
+        return this._store.select;
+      }
+    }
+
+    const component: TestComponent = TestBed.get(TestComponent);
+
+    assertType(() => component.A1$); // $ExpectType Observable<string[]>
```

Comment:
I don't think these are actually testing anything other than the return type declaration of the public property type on the class... not NGXS.